src/framebuffer.ml

```diff
@@ -1,5 +1,8 @@
 
-type t = int array array
+type t = {
+  buffer : int array array;
+  mutable dirty : bool;
```

Comment:
We should provide an accessor to read the dirty bit so that in unit tests that draw to the framebuffer we can check that it has been set.